### ISSUE TITLE
Importing json files works but typescript compiler logs error

### ISSUE THREAD

**Issue by dballesteros7** [frankwallis/plugin-typescript#129]
This code (or something similar):

```ts
import localeEn from './locale-en.json!';

export default function translateSources() {
      let en = localeEn;
}
```

Works and transpiles correctly to something like:

```js
(function(System, SystemJS) {System.register(['./locale-en.json!'], function(exports_1, context_1) {
    "use strict";
    var __moduleName = context_1 && context_1.id;
    var locale_en_json_1;
    function translateSources($translateProvider) {
        $translateProvider.translations('en', locale_en_json_1.default);
    }
    exports_1("default", translateSources);
    return {
        setters:[
            function (locale_en_json_1_1) {
                locale_en_json_1 = locale_en_json_1_1;
            }],
        execute: function() {
            translateSources.$inject = ['$translateProvider'];
        }
    }
});

})(System, System);
```

But I get console output like:

```log
TypeScript Cannot find module './locale-en.json!'. (TS2307)
```

My config file has:
```json
{
        "*.json": {
          "loader": "json"
        }
}
```

Am I doing something wrong? This is with 4.0.16. Thanks!

**Comment by frankwallis** [comment]
This is an issue with the typescript compiler, it is being addressed under https://github.com/Microsoft/TypeScript/issues/6615 (also see the issues linked to from there)

Essentially you currently need to declare modules for all these files to make the errors go away
e.g.
```
declare module './locale-en.json!' {
  var __a__: any;
  export = a;
  export default a;
}
```
(you may also need to map the path to be non-relative, I can't quite remember)

**Comment by aluanhaddad** [comment]
You can also create a file named `locale-en.json.ts` beside the json file and simply give it the content
```typescript
export default '';
```
Note that you don't need the!

**Comment by dballesteros7** [comment]
Thanks both, that sounds good! :+1: I guess I'll create a snippet for generating those files (I got one for each component :stuck_out_tongue:)

**Comment by dballesteros7** [comment]
@aluanhaddad Actually, that didn't work for me :confused:. It loads the JSON file fine but the TS file is not found despite being next to it.

**Comment by aluanhaddad** [comment]
@dballesteros7 how are you importing the file? You need to remove the `!` from the import path. 
Having configured
```JavaScript
"*.json": {
          "loader": "json"
        }
```
Means you don't need it.

**Comment by dballesteros7** [comment]
Yeah, I removed the ! and the importing works.

**Comment by aluanhaddad** [comment]
Great.

**Comment by dballesteros7** [comment]
The compiler stills complains though, even when I added the dummy .ts files.

**Comment by aluanhaddad** [comment]
Can you post some code?

**Comment by dballesteros7** [comment]
An example of what I have, 3 files in the same directory:

locales.ts
```ts
import localeEn from './locale-en.json';

export default function translateSources() {
  let en = localeEn;
 return en;
}
```
locale-en.json
```json
{
  "HOME": {
    "WELCOME_USER": "Welcome back!",
    "LOADING_USER_INFO": "Loading user information...",
    "LOGOUT": "Logout"
  }
}
```
locale-en.json.ts
```ts
export default '';
```

JSON file is loaded and the translations properly recorded but the compiler stills outputs the message:

```
logger.js:17 TypeScript Cannot find module './locale-en.json'. (TS2307)
```

**Comment by aluanhaddad** [comment]
@dballesteros7 I ran your code and you need to pass a `module` option to the TypeScript compiler. For example: `>tsc ./locales.ts --module 'system'`. If you are using a tsconfig.json file, I strongly recommend this, specify it under the compiler options property.

**Comment by aluanhaddad** [comment]
Note that in general, using modules in TypeScript without specifying a value for `module` has very odd behavior. In this case, any supported value will do, I simply use `'system'` as an example.

**Comment by frankwallis** [comment]
https://github.com/Microsoft/TypeScript/issues/6615 has been implemented and is available in plugin-typescript@5.0.0.
